lib/awesome_print/formatters/fallback_formatter.rb

```diff
@@ -0,0 +1,55 @@
+require_relative 'base_formatter'
+
+# this handles some fallback logic to route things we don't know what they are
+
+module AwesomePrint
+  module Formatters
+    class FallbackFormatter < BaseFormatter
+
+      formatter_for :self
+
+      def self.formattable?(object)
+        true
+      end
+
+      def self.core?
+        true
+      end
+
+      def format(object)
+        if @options[:raw] && object.instance_variables.any?
+          Formatters::ObjectFormatter.new(@inspector).format(object)
+        elsif (hash = convert_to_hash(object))
+          Formatters::HashFormatter.new(@inspector).format(hash)
+        else
+          Formatters::SimpleFormatter.new(@inspector).format(object.inspect.to_s)
+        end
+      end
+
+
+      private
+
+      # Utility methods.
+      #------------------------------------------------------------------------------
+      # FIXME: this could be super fixed.
+      #
+      def convert_to_hash(object)
```

Comment:
Nice to have like this:-

```suggestion
      def convert_to_hash(object)
```
     def convert_to_hash(object)
        return nil if !object.respond_to?(:to_hash)

        return nil if object.method(:to_hash).arity != 0

        hash = object.to_hash
        return nil if !hash.respond_to?(:keys) || !hash.respond_to?('[]')

        hash
      end